### PR TITLE
Sun set snowplow tracking for task history

### LINKED ISSUE #31156: Remove `task` snowplow event
The `task` Snowplow event is consuming too much computing power and we got enough data to evaluate performance gains.

We shuold remove it from the product to avoid overusage charges.

### PR DESCRIPTION
**Pull request by qnkhuat**
Closes https://github.com/metabase/metabase/issues/31156
Tracking snowplow for task history was introduced in #23893, in this PR we remove that tracking but let's still keep the schema in case we need it in the future.

### CODE REVIEW

**Review by calherries — APPROVED** [review]
Looks good, you just need to remove this reference to the ::snowplow/new-task-history event

https://github.com/metabase/metabase/blob/91a9ba1859e10be0275fcc0dbee860b55c0e4a58/test/metabase/sync/sync_metadata/sync_database_type_test.clj#L21

**Comment by nemanjaglumac** [comment]
@qnkhuat There are some related E2E failures that need to be solved before merging. Do you need help with those?

**Comment by qnkhuat** [comment]
@calherries which reference?

**Comment by qnkhuat** [comment]
Thanks @nemanjaglumac, but I fixed it myself, would be good to have a reviews tho.

File: e2e/test/scenarios/onboarding/setup/setup.cy.spec.js [hunk 1]
```diff
@@ -287,19 +287,20 @@ describeWithSnowplow("scenarios > setup", () => {
   });
 
   it("should send snowplow events", () => {
-    // 1 - pageview
+    // 1 - new_instance_created
```
Comment (qnkhuat) [inline e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:290]: see https://github.com/metabase/metabase/pull/23893#discussion_r921029709 for why we made this change in the first place

Comment (nemanjaglumac) [reply e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:290]: Gotcha, thanks!

**Review by nemanjaglumac — APPROVED** [review]
The test fix/changes make sense to me.
Thanks for the detailed explanation

**Comment by calherries** [comment]
@qnkhuat Sorry, I pasted the wrong link. I meant to suggest deleting this line: https://github.com/metabase/metabase/blob/91a9ba1859e10be0275fcc0dbee860b55c0e4a58/src/metabase/analytics/snowplow.clj#L205

**Comment by calherries** [comment]
Hmm, maybe we should delete the schema too then? It's in git if we _really_ need it.

@luizarakaki what's the likelihood of restoring task tracking again? If it's anything less than 90% I'd recommend removing the dead code for now.

**Comment by luizarakaki** [comment]
I think it can comeback, but not sure when and how. We will likely need to modify the event before re-adding it.

**Pull request merged**